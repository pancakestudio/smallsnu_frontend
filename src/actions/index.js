import * as types from './ActionTypes';

export function buildingClick(e){
  return{
    type: types.BUILDING_CLICK,
    e
  };
}
