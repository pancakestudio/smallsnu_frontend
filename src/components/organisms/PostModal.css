.postModal {
  min-width: 80vw;
}

.postModal .title {
  margin: auto 0px;
}
