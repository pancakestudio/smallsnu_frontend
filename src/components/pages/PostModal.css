.postModal {
  min-width: 80vw;
}

.postModal .title {
  margin: auto 0px;
}

.postModal .postHeader {
  display: flex;
  font-size: calc(10px + 0.7vmin);
  margin: -16px;
  padding: 5px;
}

@media(max-width: 576px){
  .postModal .postHeader {
    overflow-x: hidden;
    margin: -16px;
  }
}

.postModal .headerLeft {
  margin: auto;
}

.postModal .headerRight {
  margin: auto;
}

@media(max-width: 576px){
  .postModal .headerLeft {
    margin: 5px 0px;
  }

  .postModal .headerRight {
    margin: 2px 10px;
  }
}

.postModal .postHeader Button {
  margin: 0 0 0 5px;
  font-size: calc(10px + 0.5vmin);
  padding: 5px 7px;
}

.postModal .postContent {
  margin-top: 30px;
}

.postModal .like {
  min-width: 50px;
  border-color: #FF007F;
  color: #FF007F;
  margin-top: 60px;
}

.postModal .like:hover {
  border-color: #FF007F;
  color: white;
  background-color: #FF007F;
}

.postModal .like:focus {
  border-color: #FF007F;
  color: white;
  background-color: #FF007F !important;
}

.postModal .like:active {
  border-color: #FF007F;
  color: white;
  background-color: #FF007F !important;
}

.postModal .commentsHeader {
  background-color: rgba(0,0,0,0);
  margin: 0 -16px -16px -16px;
}

.postModal .commentsBody {
  margin: 0 -16px -16px -16px; 
}

.postModal .commentIcon {
  margin: auto 4px 1px auto;
}

.postModal .writeComment {
  width: 100%;
  margin-bottom: -16px;
}

.postModal .commentsBody .commenter {
  margin-left: -16px;
  font-size: calc(10px + 0.7vmin);
}

.postModal .commentsBody .configure {
  font-size: calc(10px + 0.7vmin);
}

.postModal .commentsBody .configure Button {
  margin-right: -32px;
  font-size: calc(10px + 0.5vmin);
  padding: 0.5px 7px;
}

.postModal .commentsBody .dropdown-menu {
  min-width: 60px;
  text-align: center;
}

.postModal .commentsBody .deleteComment {
  color: red;
}

.postModal .commentsBody .likeComment {
  border-color: rgba(0,0,0,0);
  color: #FF007F;
  background-color: rgba(0,0,0,0);
  font-size: calc(10px + 0.5vmin);
  padding: 2px 0px;
}

.postModal .commentsBody .likeComment:focus {
  color: #FFC0CB !important;
  background-color: rgba(0,0,0,0) !important;
  border-color: rgba(0,0,0,0) !important;
}

.postModal .commentsBody .likeComment:active {
  color: #FFC0CB !important;
  background-color: rgba(0,0,0,0) !important;
  border-color: rgba(0,0,0,0) !important;
}
