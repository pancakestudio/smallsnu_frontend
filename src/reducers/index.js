import { combineReducers } from 'redux';
import buildingButton from './buildingButton';

const reducers = combineReducers({
  buildingButton
});

export default reducers;
