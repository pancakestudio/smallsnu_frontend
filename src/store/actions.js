import * as types from './actionTypes'

/******** Map ********/

export function requestBuilding(bldgNo){
  return{
    type: types.REQUEST_BUILDING,
    bldgNo
  };
}

export function getBuildingSuccess(krName, bldgNo, info, restaurants, seminars, lectures, posts){
  return{
    type: types.GET_BUILDING_SUCCESS,
    krName,
    bldgNo,
    info,
    restaurants,
    seminars,
    lectures,
    posts,
  }
}

export function getBuildingFailure(error){
  return{
    type: types.GET_BUILDING_FAILURE,
    error
  }
}

export function zoomChanged(zoomLevel){
  return {
    type: types.ZOOM_CHANGED,
    zoomLevel
  }
}

export function hideMarkers(){
  return {
    type: types.HIDE_MARKERS
  }
}

/******** SideBar ********/

export function toggleSidebar(){
  return {
    type: types.TOGGLE_SIDE_BAR
  }
}

export function toggleShuttleMenuMarker(){
  return{
    type: types.TOGGLE_SHUTTLE_MENU_MARKER
  }
}

/******** SearchBar ********/

export function search(bldgNo, bldgPos){
  return{
    type: types.SEARCH,
    bldgNo,
    bldgPos
  }
}

/******** Shuttle ********/
export function requestAllShuttles(){
  return{
    type: types.REQUEST_ALL_SHUTTLE
  }
}

export function getAllShuttleSuccess(stations){
  return{
    type: types.GET_ALL_SHUTTLE_SUCCESS,
    stations
  }
}

export function toggleShuttleMarker(){
  return{
    type: types.TOGGLE_SHUTTLE_MARKER
  }
}

export function requestAllRevShuttles(){
  return{
    type: types.REQUEST_ALL_REV_SHUTTLE
  }
}

export function getAllRevShuttleSuccess(stations){
  return{
    type: types.GET_ALL_REV_SHUTTLE_SUCCESS,
    stations
  }
}

export function toggleRevShuttleMarker(){
  return{
    type: types.TOGGLE_REV_SHUTTLE_MARKER
  }
}

/******** Post ********/

export function requestBoard(bldgNo){
  return {
    type: types.REQUEST_BOARD,
    bldgNo
  }
}

export function getBoardSuccess(posts, bldgNo){
  return {
    type: types.GET_BOARD_SUCCESS,
    posts,
    bldgNo
  }
}

export function getBoardFailure(error){
  return {
    type: types.GET_BOARD_FAILURE,
    error
  }
}

export function requestPost(id) {
  return {
    type: types.REQUEST_POST,
    id
  }
}

export function getPostSuccess(post){
  return {
    type: types.GET_POST_SUCCESS,
    post
  }
}

export function getPostFailure(error){
  return {
    type: types.GET_POST_FAILURE,
    error
  }
}

export function savePost(post, bldgNo){
  return{
    type: types.SAVE_POST,
    post,
    bldgNo
  }
}

export function savePostSuccess(bldgNo){
  return {
    type: types.SAVE_POST_SUCCESS,
    bldgNo
  }
}

export function savePostFailure(error){
  return {
    type: types.SAVE_POST_FAILURE,
    error
  }
}

export function editPost(post, bldgNo){
  return{
    type: types.EDIT_POST,
    post,
    bldgNo,
  }
}

export function editPostSuccess(postId){
  return {
    type: types.EDIT_POST_SUCCESS,
    postId
  }
}

export function editPostFailure(error){
  return {
    type: types.EDIT_POST_FAILURE,
    error
  }
}

export function editPostFlag(){
  return{
    type: types.EDIT_POST_FLAG,
  }
}

export function deletePost(post, bldgNo, password){
  return{
    type: types.DELETE_POST,
    post,
    bldgNo,
    password
  }
}

export function deletePostSuccess(bldgNo){
  return {
    type: types.DELETE_POST_SUCCESS,
    bldgNo
  }
}

export function deletePostFailure(error){
  return {
    type: types.DELETE_POST_FAILURE,
    error
  }
}

export function showWritePost(bldgNo) {
  return{
    type: types.SHOW_WRITE_POST,
    bldgNo
  };
}

export function hideWritePost() {
  return {
    type: types.HIDE_WRITE_POST,
  }
}

export function showPasswordCheck(target){
  return{
    type: types.SHOW_PASSWORD_CHECK,
    target
  }
}

export function hidePasswordCheck(){
  return{
    type: types.HIDE_PASSWORD_CHECK
  }
}

export function postLike(postId) {
  return {
    type: types.POST_LIKE,
    postId
  }
}

export function postLikeSuccess(postId) {
  return {
    type: types.POST_LIKE_SUCCESS,
    postId
  }
}

export function postLikeFailure(error) {
  return {
    type: types.POST_LIKE_FAILURE,
    error
  }
}

export function saveComment(comment, postId){
  return {
    type: types.SAVE_COMMENT,
    comment,
    postId
  }
}

export function saveCommentSuccess(postId){
  return {
    type: types.SAVE_COMMENT_SUCCESS,
    postId
  }
}

export function saveCommentFailure(error){
  return {
    type: types.SAVE_COMMENT_FAILURE,
    error
  }
}

export function showEditComment(comment){
  return {
    type: types.SHOW_EDIT_COMMENT,
    comment
  }
}

export function hideEditComment(){
  return {
    type: types.HIDE_EDIT_COMMENT
  }
}

export function editComment(comment, postId){
  return {
    type: types.EDIT_COMMENT,
    comment,
    postId
  }
}

export function editCommentSuccess(postId){
  return {
    type: types.EDIT_COMMENT_SUCCESS,
    postId
  }
}

export function editCommentFailure(error){
  return {
    type: types.EDIT_COMMENT_FAILURE,
    error
  }
}

export function deleteComment(comment, postId, password){
  return {
    type: types.DELETE_COMMENT,
    comment,
    postId,
    password
  }
}

export function deleteCommentSuccess(msg){
  return {
    type: types.DELETE_COMMENT_SUCCESS,
    msg
  }
}

export function deleteCommentFailure(error){
  return {
    type: types.DELETE_COMMENT_FAILURE,
    error
  }
}

export function commentLike(commentId, postId) {
  return {
    type: types.COMMENT_LIKE,
    commentId,
    postId
  }
}

export function commentLikeSuccess(postId) {
  return {
    type: types.COMMENT_LIKE_SUCCESS,
    postId
  }
}

export function commentLikeFailure(error) {
  return {
    type: types.COMMENT_LIKE_FAILURE,
    error
  }
}

export function changeBoardPage(page){
  return {
    type: types.CHANGE_BOARD_PAGE,
    page
  }
}

export function wrongPassword(){
  return {
    type: types.WRONG_PASSWORD,
  }
}

/******** Restaurant ********/

export function toggleResMarker(){
  return{
    type: types.TOGGLE_RES_MARKER,
  }
}

export function requestRestaurant(id){
  return{
    type: types.REQUEST_RESTAURANT,
    id
  }
}

export function getRestaurantSuccess(restaurant){
  return{
    type : types.GET_RESTAURANT_SUCCESS,
    restaurant,
  }
}

export function getRestaurantFailure(error){
  return{
    type: types.GET_RESTAURANT_FAILURE,
    error
  }
}

export function requestAllRestaurants(){
  return{
    type: types.REQUEST_ALL_RESTAURANTS,
  }
}


export function getAllRestaurantsSuccess(restaurants){
  return{
    type : types.GET_ALL_RESTAURANTS_SUCCESS,
    restaurants,
  }
}

export function getAllRestaurantsFailure(error){
  return{
    type: types.GET_ALL_RESTAURANTS_FAILURE,
    error
  }
}

/******** Seminar ********/

export function toggleSemiMarker(){
  return{
    type: types.TOGGLE_SEMI_MARKER,
  }
}

export function changeSeminarPage(page){
  return {
    type: types.CHANGE_SEMINAR_PAGE,
    page
  }
}

export function requestSeminar(id){
  return {
    type: types.REQUEST_SEMINAR,
    id
  }
}

export function getSeminarSuccess(seminar){
  return {
    type: types.GET_SEMINAR_SUCCESS,
    seminar
  }
}

export function getSeminarFailure(error){
  return {
    type: types.GET_SEMINAR_FAILURE,
    error
  }
}

export function requestBldgSeminars(bldgNo){
  return {
    type: types.REQUEST_BLDG_SEMINARS,
    bldgNo
  }
}

export function getBldgSeminarsSuccess(seminars, bldgNo){
  return {
    type: types.GET_BLDG_SEMINARS_SUCCESS,
    seminars,
    bldgNo
  }
}

export function getBldgSeminarsFailure(error){
  return {
    type: types.GET_BLDG_SEMINARS_FAILURE,
    error
  }
}

export function requestAllSeminars(){
  return {
    type: types.REQUEST_ALL_SEMINARS,
  }
}

export function getAllSeminarsSuccess(seminars){
  return {
    type: types.GET_ALL_SEMINARS_SUCCESS,
    seminars
  }
}

export function getAllSeminarsFailure(error){
  return {
    type: types.GET_ALL_SEMINARS_FAILURE,
    error
  }
}

/******** Bank ********/

export function toggleBankMarker(){
  return{
    type: types.TOGGLE_BANK_MARKER,
  }
}

export function requestAllBanks(){
  return{
    type: types.REQUEST_ALL_BANKS,
  }
}

export function getAllBanksSuccess(banks){
  return{
    type: types.GET_ALL_BANKS_SUCCESS,
    banks,
  }
}

export function getAllBanksFailure(error){
  return{
    type: types.GET_ALL_BANKS_FAILURE,
    error,
  }
}

export function requestBank(id){
  return{
    type: types.REQUEST_BANK,
    id
  }
}

export function getBankSuccess(bank){
  return{
    type: types.GET_BANK_SUCCESS,
    bank
  }
}

export function getBankFailure(error){
  return{
    type: types.GET_BANK_FAILURE,
    error
  }
}


/******** ATM ********/

export function toggleATMMarker(){
  return{
    type: types.TOGGLE_ATM_MARKER,
  }
}

export function requestAllATMs(){
  return{
    type: types.REQUEST_ALL_ATMS,
  }
}

export function getAllATMsSuccess(atms){
  return{
    type: types.GET_ALL_ATMS_SUCCESS,
    atms,
  }
}

export function getAllATMsFailure(error){
  return{
    type: types.GET_ALL_ATMS_FAILURE,
    error,
  }
}

export function requestATM(id){
  return{
    type: types.REQUEST_ATM,
    id
  }
}

export function getATMSuccess(atm){
  return{
    type: types.GET_ATM_SUCCESS,
    atm
  }
}

export function getATMFailure(error){
  return{
    type: types.GET_ATM_FAILURE,
    error
  }
}

/******** Cafe ********/

export function toggleCafeMarker(){
  return{
    type: types.TOGGLE_CAFE_MARKER,
  }
}

export function requestAllCafes(){
  return{
    type: types.REQUEST_ALL_CAFES,
  }
}

export function getAllCafesSuccess(cafes){
  return{
    type: types.GET_ALL_CAFES_SUCCESS,
    cafes,
  }
}

export function getAllCafesFailure(error){
  return{
    type: types.GET_ALL_CAFES_FAILURE,
    error,
  }
}

export function requestCafe(id){
  return{
    type: types.REQUEST_CAFE,
    id
  }
}

export function getCafeSuccess(cafe){
  return{
    type: types.GET_CAFE_SUCCESS,
    cafe
  }
}

export function getCafeFailure(error){
  return{
    type: types.GET_CAFE_FAILURE,
    error
  }
}

/******** Conv ********/

export function toggleConvMarker(){
  return{
    type: types.TOGGLE_CONV_MARKER,
  }
}

export function requestAllConves(){
  return{
    type: types.REQUEST_ALL_CONVES,
  }
}

export function getAllConvesSuccess(conves){
  return{
    type: types.GET_ALL_CONVES_SUCCESS,
    conves,
  }
}

export function getAllConvesFailure(error){
  return{
    type: types.GET_ALL_CONVES_FAILURE,
    error,
  }
}

export function requestConv(id){
  return{
    type: types.REQUEST_CONV,
    id
  }
}

export function getConvSuccess(conv){
  return{
    type: types.GET_CONV_SUCCESS,
    conv
  }
}

export function getConvFailure(error){
  return{
    type: types.GET_CONV_FAILURE,
    error
  }
}


/******** Main ********/

export function refreshData(){
  return {
    type: types.REFRESH_DATA
  }
}
