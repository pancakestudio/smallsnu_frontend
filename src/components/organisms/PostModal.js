import React from 'react'
import { Modal, Button } from 'react-bootstrap'
import { historyPush } from '../../utils/Functions'
import './PostModal.css'

export const PostModal = ({post}) => {
  const handleBack = () => {
    if(post && post.building){
      let boardNo = post.building.code
      historyPush(`/board/${boardNo}`)
    } else {
      historyPush('/')
    }
  }

  let modal

  if(post){
    modal = <Modal
      show = {true}
      onHide = {()=>{historyPush('/')}}
      dialogClassName = "postModal"
      scrollable = {true}
      centered
    >
      <Modal.Header closeButton>
        <Modal.Title>{post.title}</Modal.Title>
      </Modal.Header>

      <Modal.Body>
        {post.content}
      </Modal.Body>

      <Modal.Footer>
        <Button variant = "secondary" onClick = {() =>handleBack()}>뒤로가기</Button>
      </Modal.Footer>

    </Modal>
  }else{
    modal = <Modal
      show = {true}
      onHide = {()=>{historyPush('/')}}
      centered
    >
      <Modal.Header closeButton>
        <Modal.Title>게시글 정보가 없습니다.</Modal.Title>
      </Modal.Header>
  </Modal>
  }
  return modal
}
