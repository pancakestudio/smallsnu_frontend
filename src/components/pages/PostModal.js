import React from 'react'
import moment from 'moment'
import { Modal, Button, Dropdown, DropdownButton, Card, ListGroup, Container, Row, Col } from 'react-bootstrap'
import { FaAngleLeft, FaHeart, FaRegComment } from 'react-icons/fa'
import { ModalSpinner } from '../molecules/ModalSpinner'
import WriteComment from '../../containers/WriteComment'
import { historyPush, trimCreated } from '../../utils/Functions'
import './PostModal.css'

export const PostModal = ({post, onShowWritePostModal, onEdit, onDelete, onLike, onEditComment, onDeleteComment, onLikeComment}) => {
  const handleBack = () => {
    if(post && post.building){
      const boardNo = post.building.code
      historyPush(`/board/${boardNo}`)
    }
  }
  const handleEdit = () => {
    onEdit(post, post.building.code)
  }
  const handleDelete = () => {
    onDelete(post)
  }
  const handleLike = () => {
    let liked = localStorage.getItem("likedPosts")
    liked = (liked) ? JSON.parse(liked) : []
    if(liked.includes(post.id)){
      alert('이미 추천한 게시글입니다.')
    } else {
      onLike(post.id)
      liked.push(post.id)
      localStorage.setItem("likedPosts", JSON.stringify(liked))
    }
  }
  const handleEditComment = (comment) => {
    onEditComment(comment)
  }
  const handleDeleteComment = (comment) => {
    onDeleteComment(comment)
  }
  const handleLikeComment = (commentId) => {
    let liked = localStorage.getItem("likedComments")
    liked = (liked) ? JSON.parse(liked) : []
    if(liked.includes(commentId)){
      alert('이미 추천한 댓글입니다.')
    } else {
      onLikeComment(commentId, post.id)
      liked.push(commentId)
      localStorage.setItem("likedComments", JSON.stringify(liked))
    }
  }

  let comments, body
  if(post && post.comments){
    if(post.comments.length!==0){
      body = 
      <Card.Body className="commentsBody">
        <ListGroup variant="flush">
          {post.comments.map(com => (
            <ListGroup.Item key={com.id}> 
              <Container fluid>
                <Row>
                  <Col xs={10} md={10} className="commenter text-muted">
                    {com.username} | {trimCreated(com.created)}<br/>
                  </Col>
                  <Col xs={2} md={2} className="configure text-right">
                    <DropdownButton
                      key={com.id}
                      title=""
                      alignRight
                      variant="outline-secondary"
                      drop="down"
                    >
                      <Dropdown.Item
                        as="button"
                        className="editComment"
                        onClick={()=>{handleEditComment(com)}}
                      >
                        수정
                      </Dropdown.Item>
                      <Dropdown.Item
                        as="button"
                        className="deleteComment"
                        onClick={()=>{handleDeleteComment(com)}}
                      >
                        삭제
                      </Dropdown.Item>
                    </DropdownButton>
                  </Col>
                </Row>
                <Row>
                  {com.content}<br/>
                </Row>
                <Row>
                  <Button
                    className="likeComment"
                    onClick={()=>handleLikeComment(com.id)}
                  >
                    <FaHeart /> {com.like}
                  </Button>
                </Row>
              </Container>
            </ListGroup.Item>
          ))}
        </ListGroup>
      </Card.Body>
    }

    comments =
    <Card className="comments border-left-0 border-right-0 border-bottom-0">
      <Card.Header className="commentsHeader">
        <FaRegComment className="commentIcon"/>{post.comments.length}
      </Card.Header>
      {body}
    </Card>
  }

  if(post && Object.keys(post).length>0){
    return (
      <Modal
        show = {true}
        onHide = {()=>{historyPush('/')}}
        dialogClassName = "postModal"
        scrollable = {true}
        centered
      >
        <Modal.Header closeButton>
          <Button className="back" onClick={handleBack}><FaAngleLeft /></Button>
          <Modal.Title className="title">{post.title}</Modal.Title>
        </Modal.Header>

        <Modal.Body>
          <Card className="mainText border-0">
            <Card.Header className="postHeader text-muted">
              <Container fluid>
              <Row>
                <Col xs={12} md={8} className="headerLeft">
                  작성자: {post.username} | 작성일시: {moment(post.created).format("YYYY.MM.DD HH:mm:ss")}
                </Col>
                <Col xs={12} md={4} className="headerRight text-right">
                  <Button variant = "secondary" onClick={handleEdit}>수정</Button>
                  <Button variant = "danger" onClick={handleDelete}>삭제</Button>
                </Col>
              </Row>
              </Container>
            </Card.Header>
            <Card.Body>
              <Card.Text className="postContent">
                {post.content}
              </Card.Text>
              <div className="text-center">
                <Button
                  variant="outline-danger"
                  className="like text-center"
                  onClick={handleLike}
                >
                  <FaHeart/><br/>{post.like}
                </Button>
              </div>
            </Card.Body>
          </Card>
          { comments }
        </Modal.Body>
        <Modal.Footer className="postFooter">
          <WriteComment className="writeComment" postId={post.id}/>
        </Modal.Footer>
      </Modal>
    )
  }else{
    return (
      <ModalSpinner />
    )
  }
}
