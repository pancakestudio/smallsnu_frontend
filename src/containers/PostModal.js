import { connect } from 'react-redux'
import { PostModal } from '../components/pages/PostModal'
import { showWritePost , editPostFlag, showPasswordCheck, postLike, showEditComment, commentLike } from '../store/actions'

const mapStateToProps = (state) => {
  return {
    post: state.selectedPost,
  }
}

const mapDispatchToProps = (dispatch) => {
  return{
    onShowWritePostModal: (bldgNo) => {
      dispatch(showWritePost(bldgNo))
    },
    onEdit: () => {
      dispatch(editPostFlag())
    },
    onDelete: (post) => {
      dispatch(showPasswordCheck(post))
    },
    onLike: (postId) => {
      dispatch(postLike(postId))
    },
    onEditComment: (comment) => {
      dispatch(showEditComment(comment))
    },
    onDeleteComment: (comment) => {
      dispatch(showPasswordCheck(comment))
    },
    onLikeComment: (commentId, postId) => {
      dispatch(commentLike(commentId, postId))
    }
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(PostModal)
