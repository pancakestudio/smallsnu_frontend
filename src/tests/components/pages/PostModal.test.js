import React from 'react'
import { Provider } from 'react-redux'
import { PostModal } from '../../../components/pages/PostModal'
import ConnectedPostModal from '../../../containers/PostModal'
import configureStore from 'redux-mock-store'
import { shallow, mount } from 'enzyme'
import reducers from '../../../store/reducers'
import { createStore } from 'redux'
import * as actions from '../../../store/actions'

describe('PostModal', () => {
  let component
  const mockShowWritePostModal = jest.fn()
  const mockEdit = jest.fn()
  const mockDelete = jest.fn()
  const mockLike = jest.fn()
  const mockEditComment = jest.fn()
  const mockDeleteComment = jest.fn()
  const mockLikeComment = jest.fn()
  global.window = { location: { pathname: '/post/1' } };

  it('renders correctly', () => {
    component = shallow(
    <PostModal
      post = {{
        "id" : 1,
        "username" : "user1",
        "title": "post1",
        "content": "post content 1",
        "building":{
          "code":"301",
        },
        "comments": [
          {
            "id" : 1,
            "content": "comment1",
            "username": "user1"
          },
          {
            "id" : 2,
            "content": "comment2",
            "username": "user2"
          },
        ]
      }}
      onShowWritePostModal = {mockShowWritePostModal}
      onEdit = {mockEdit}
      onDelete = {mockDelete}
      onLike = {mockLike}
      onEditComment = {mockEditComment}
      onDeleteComment = {mockDeleteComment}
      onLikeComment = {mockLikeComment}
    />
    )
  })

  it('matches snapshot', ()=>{
    expect(component).toMatchSnapshot()
  })

  it('has a modal', ()=>{
    expect(component.find('Bootstrap(Modal)').exists()).toBe(true)
  })

  it('calls functions', ()=>{
    component.find('Button').at(1).simulate('click')
    expect(mockEdit.mock.calls.length).toBe(1)
    component.find('Button').at(2).simulate('click')
    expect(mockDelete.mock.calls.length).toBe(1)
    component.find('.like').simulate('click')
    expect(mockLike.mock.calls.length).toBe(1)
    component.find('.editComment').at(0).simulate('click')
    expect(mockEditComment.mock.calls.length).toBe(1)
    component.find('.deleteComment').at(0).simulate('click')
    expect(mockDeleteComment.mock.calls.length).toBe(1)
    component.find('.likeComment').at(0).simulate('click')
    expect(mockLikeComment.mock.calls.length).toBe(1)
    component.find('.back').simulate('click')
    expect(global.window.location.pathname).toEqual('/board/301')
    component.find('Bootstrap(Modal)').simulate('hide')
    expect(global.window.location.pathname).toEqual('/')
  })

  it('handles undefined data correctly', ()=>{
    component.setProps({post: undefined})
    expect(component.find('ModalSpinner').exists()).toBe(true)
  })
})

describe('ConnectedPostModal', () => {
  const post = {
    "id" : "2",
    "username" : "user1",
    "password" : "1234",
    "title": "post1",
    "content": "post content 1",
    "building":{
      "code":"301",
    },
    "comments": [
      {
        "id" : "3",
        "content": "comment1",
        "username": "user1"
      },
      {
        "id" : "4",
        "content": "comment2",
        "username": "user2"
      },
    ]
  }
  const initialState = {
    selectedPost: post
  }
  const mockStore = configureStore()
  let store, component

  it('renders correctly', ()=>{
    store = mockStore(initialState)
    component = mount(<Provider store={store}><ConnectedPostModal/></Provider>)
  })

  it('matches snapshot', ()=>{
    expect(component).toMatchSnapshot()
  })

  it('has a modal', ()=>{
    expect(component.find('Bootstrap(Modal)').exists()).toBe(true)
  })

  it('dispatches showWritePostModal action and editPostFlag action correctly', ()=>{
    component.find('Button').at(1).prop('onClick')()
    expect(store.getActions()[0]).toEqual(actions.showEditPost(post))
  })

  it('dispatches showPasswordCheck action correctly', ()=>{
    component.find('Button').at(2).prop('onClick')()
    expect(store.getActions()[1]).toEqual(actions.showPasswordCheck(post))
  })

  it('dispatches postLike action correctly', ()=>{
    component.find('Button').at(3).prop('onClick')()
    expect(store.getActions()[2]).toEqual(actions.postLike('2'))
  })

  it('dispatches showEditComment action correctly', ()=>{
    component.find('DropdownToggle').at(0).simulate('click')
    component.find('.editComment').at(0).prop('onClick')()
    expect(store.getActions()[3]).toEqual(actions.showEditComment(
      {
        "id" : "3",
        "content": "comment1",
        "username": "user1"
      }
    ))
  })

  it('dispatches comment showPasswordCheck action correctly', ()=>{
    component.find('.deleteComment').at(0).prop('onClick')()
    expect(store.getActions()[4]).toEqual(actions.showPasswordCheck(
      {
        "id" : "3",
        "content": "comment1",
        "username": "user1"
      }
    ))
  })

  it('dispatches commentLike action correctly', ()=>{
    component.find('.likeComment').at(0).prop('onClick')()
    expect(store.getActions()[5]).toEqual(actions.commentLike('3', '2'))
  })
})
