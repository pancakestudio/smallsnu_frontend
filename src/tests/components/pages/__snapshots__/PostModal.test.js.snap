// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectedPostModal matches snapshot 1`] = `ReactWrapper {}`;

exports[`PostModal matches snapshot 1`] = `ShallowWrapper {}`;
